eth/v1alpha1/validator.proto

```diff
@@ -15,13 +15,119 @@ syntax = "proto3";
 
 package ethereum.eth.v1alpha1;
 
+import "google/protobuf/empty.proto";
+import "eth/v1alpha1/block.proto";
+
 option csharp_namespace = "Ethereum.Eth.v1alpha1";
 option go_package = "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1;eth";
 option java_multiple_files = true;
 option java_outer_classname = "ValidatorProto";
 option java_package = "org.ethereum.eth.v1alpha1";
 option php_namespace = "Ethereum\\Eth\\v1alpha1";
 
+// Beacon node validator API
+//
+// The beacon node validator API enables a validator to connect
+// and perform its obligations on the Ethereum 2.0 phase 0 beacon chain.
+service BeaconNodeValidator {
+    // Retrieves validator duties for the requested validators.
+    //
+    // The server returns a list of duties which are the actions should be performed by validators for a given epoch.
+    // Validator duties should be polled every epoch, but due to chain reorg of >MIN_SEED_LOOKAHEAD could occur,
+    // the validator duties could chain. For complete safety, it is recommended to poll at every slot to ensure
+    // validator is fully aware of any sudden chain reorg.
+    rpc Duties(DutiesRequest) returns (DutiesResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/duties"
+        };
+    }
+
+    // Retrieves the latest valid beacon block to be proposed on the beacon chain.
+    //
+    // The server returns a new beacon block, without proposer signature, that can be
+    // proposed on the beacon chain. The block should be filled with all the necessary
+    // data for proposer to sign.
+    rpc Block(BlockRequest) returns (Block) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/block"
+        };
+    }
+
+    // Sends the newly signed beacon block to beacon node to be broadcasted to the rest of the network.
+    //
+    // The validator sends the newly signed beacon block to the beacon node so the beacon block can
+    // be included in the beacon chain. The beacon node is expected to validate and process the
+    // beacon block into its state, however if the beacon block fails validator, it will still
+    // be broadcasted to the rest of the network but a different status code is returned.
```

Comment:
What should we expect the server to return on a validation error? 

Choose a code from here: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

Reply:
Why would it broadcast an invalid block?

Reply:
If an incoming block failed to process doesn't mean it fails to every node in the network. Every node should get a fair chance to process it. This behavior was reviewed and agreed on with Lighthouse and Danny

We will have these 2 status code to differentiate:
1. The block was validated successfully and has been broadcast. It has also been integrated into receiving beacon node's database.
2. The block failed validation, but was successfully broadcast anyway. It was not integrated into the beacon node's database.

Same for attestation

Reply:
Can you elaborate more on the rationale? I think this is very risky to broadcast messages that you know will fail. 

This seems like a great way to spam the network with any public beacon chain endpoint. You could send junk attestations/blocks. Peers may disconnect you if you are sending data that is bad or cannot be processed.

I generally disagree with this idea and do not think we should include this detail in this schema definition.

Reply:
We dont know if incoming block will fail for everyone else, say my node is at `state_A` and majority of the nodes are at `state_B`, and the incoming is a valid block for `state_B`. By not helping to propagate through rest of the network, we can degrade the liveness of the chain. 

Note, there's only 1 valid block proposer per slot and hopefully 2-3 well-aggregated attestations per slot, the spam thing is less of a concern. We are not propagating any block, we are only propagating blocks that coming from the correct proposer but failed state transition, I can update the comment to make it more clear

Reply:
Talked offline. Agreed with @prestonvanloon's rationale. I have removed the mention of bad block and attestation propagations

Reply:
We are continuing this idea/discussion here: https://github.com/prysmaticlabs/prysm/issues/2871